index.ts

```diff
@@ -0,0 +1,186 @@
+
+const colorPlatform = (platform: string) => {
+    switch (platform) {
+        case 'mobile':
+            return chalk.magenta(platform);
+        case 'desktop':
+            return chalk.cyan(platform);
+        default:
+            return chalk.gray(platform);
+    }
+};
+
+const localRun = (urlWithRun: string) => {
+    // TODO: allow local to simulate mobile. Will only simulate desktop.
+    const { status = -1, stdout } = spawnSync('node', [
+        lighthouseCli,
+        '--config-path=./node_modules/lighthouse/lighthouse-core/config/lr-desktop-config.js',
+        urlWithRun,
+        '--chromeFlags="--headless"',
+        '--output=json',
+        '--throttling.rttMs=40',
+        '--throttling.throughputKbps=10240',
+        // https://lighthouse-cpu-throttling-calculator.vercel.app/
+        '--throttling.cpuSlowdownMultiplier=6',
+    ]);
+    if (status !== 0) {
+        throw new Error('Lighthouse failed, skipping run...');
+    }
+    return JSON.parse(stdout.toString());
+};
+
+const psiRun = async (
+    urlWithRun: string,
+    key: string,
+    platform: (typeof platforms)[number]
+): Promise<psi.LighthouseResult> => {
+    const { data } = await psi(urlWithRun, {
+        key,
+        strategy: platform,
+        // @ts-ignore -- this is necessary for the current setup to work but does not exist in the types
+        category: ['performance', 'accessibility', 'best-practices', 'seo', 'pwa'],
+    });
+
+    return data.lighthouseResult;
+};
+
+// Lab results
+const singleOutput = (runnerResult: psi.LighthouseResult) => {
+    const { performance, seo, accessibility, 'best-practices': bestPractices } = runnerResult.categories;
+    const log = console.log;
+    log('\n', colorPlatform(runnerResult.configSettings.emulatedFormFactor ?? ''), runnerResult.finalUrl);
+    log('performance:', colorScore(processScore(performance.score)));
+    log('accessibility:', colorScore(processScore(accessibility.score)));
+    log('bestPractices:', colorScore(processScore(bestPractices.score)));
+    log('seo:', colorScore(processScore(seo.score)), '\n');
+};
+
+interface CustomResults {
+    [website: string]: {
+        [platform: string]: Array<{
+            performance: number;
+            accessibility: number;
+            'best-practices': number;
+            seo: number;
+        }>;
+    };
+}
+
+export const runPsi = async (options: Options) => {
+    console.log('Running PageSpeed Insights...');
+    const numRuns = options.num ?? NUM_RUNS;
+
+    options.mobile && platforms.filter((url) => url === 'mobile');
+
+    let customResults: CustomResults = {};
+
+    for (const url of options.websites) {
+        for (const platform of platforms) {
+            const results = [];
+            for (let i = 0; i < numRuns; i++) {
+                // To prevent Google PSI API from returning the previous cached result
+                const urlWithRun = `${url}?run=${uuid()}`;
```

Comment:
Could you talk through why you opted for a UUID here rather than `1` - `5`? I'm wondering if we can use [`randomUUID`](https://developer.mozilla.org/en-US/docs/Web/API/Crypto/randomUUID) from [`node:crypto`](https://bun.sh/docs/runtime/nodejs-apis#node-crypto) to cut down the number of dependencies needed to install.

Reply:
Mmm, that might work! I've just always used the uuid library for this type of thing. 

Since these runs are being cached by the URL, I was thinking that if it was just 1-5 even those run=x runs would be cached, so if something went wrong and we needed to do a clean run without cache we couldn't do so easily.

Now thinking about it I might actually want to add a new CLI option for this: something like `--disable-cache`? That way it could work based on the 1-5 or a UUID if we're ok with having some runs cached.

Reply:
Roger, your explanation makes sense! Let's not overcomplicate the CLI yet so no need to add the option. Let's keep the UUID but see if you can use the internal package from bun.

Reply:
Roger, will do!